src/hardware/vga_draw.cpp

```diff
@@ -353,42 +353,56 @@ static uint8_t * VGA_Draw_Linear_Line(Bitu vidstart, Bitu /*line*/) {
 
 static uint8_t* draw_linear_line_from_dac_palette(Bitu vidstart, Bitu /*line*/)
 {
-	const auto offset = vidstart & vga.draw.linear_mask;
-	const uint8_t *ret = &vga.draw.linear_base[offset];
-	const auto palette_map = vga.dac.palette_map;
+	const auto bytes_per_vga_pixel = check_cast<uint8_t>(vga.draw.bpp / 8);
+	assert(bytes_per_vga_pixel > 0);
+	const auto pixels_per_line = check_cast<uint16_t>(vga.draw.line_length /
+	                                                  bytes_per_vga_pixel);
+
+	// Quick reference
+	constexpr auto palette_map = vga.dac.palette_map;
+
+	// Ensure the size of the palettized VGA line will fit
+	constexpr uint8_t bytes_per_palette_pixel = sizeof(palette_map[0]);
+	assert(pixels_per_line * bytes_per_palette_pixel <=
+	       static_cast<int>(templine_buffer.size()));
+
+	// Running write position for the palettized VGA pixels
+	auto target_addr = TempLine;
+
+	auto write_pixels_from = [&](const uint8_t* vga_line_start,
```

Comment:
I know it will be a bit worse on the eyes, but in this case I strongly suggest capturing references to `palette_map` and `target_addr` explicitly instead of all variables in context:

Replace `[&]` with `[&palette_map, &target_addr]`.

In this case I think it's reasonable because `target_addr` is de-facto in-out parameter for this lambda.

Reply:
Looks good @dreamer; much clearer what's being scoped-in.